x-pack/plugins/fleet/public/applications/fleet/sections/agents/components/agent_upgrade_modal/index.tsx

```diff
@@ -274,7 +279,12 @@ export const AgentUpgradeAgentModal: React.FunctionComponent<AgentUpgradeAgentMo
       }
     >
       <p>
-        {isSingleAgent ? (
+        {noVersions ? (
+          <FormattedMessage
+            id="xpack.fleet.upgradeAgents.noVersionsText"
+            defaultMessage="No versions available to upgrade to. Please select agents which have upgrade available."
```

Comment:
Let's reword this to "No selected agents are eligible for an upgrade. Please select one or more eligible agents."

Reply:
I'll reword and take a closer look on your suggestion with disabling the bulk action.